### PR TITLE
SONAR-6681 Move computation of Unit test related measures to Compute Engine

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/UnitTestMeasuresStep.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+/**
+ * Computes unit test measures on files and then aggregates them on higher components.
+ */
+public class UnitTestMeasuresStep implements ComputationStep {
+
+  private final TreeRootHolder treeRootHolder;
+  private final MetricRepository metricRepository;
+  private final MeasureRepository measureRepository;
+
+  private final ImmutableList<Formula> FORMULAS = ImmutableList.<Formula>of(
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/UnitTestMeasuresStep.java]: MAJOR Rename this field "FORMULAS" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00116)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/UnitTestMeasuresStep.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        int errors = counter.testsErrorsCounter.getValue().get();
+        int failures = counter.testsFailuresCounter.getValue().get();
+        double density = (errors + failures) * 100d / tests;
+        return Optional.of(Measure.newMeasureBuilder().create(100d - density));
+      }
+      return Optional.absent();
+    }
+
+    private static boolean isPositive(Optional<Integer> value, boolean isStrictComparison) {
+      return value.isPresent() && (isStrictComparison ? value.get() > 0 : value.get() >= 0);
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/UnitTestMeasuresStep.java]: MAJOR Add parentheses to make the operator precedence explicit. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS864)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by sns-seb** [comment]
a bunch of comments and questions
also, can not be merged into master in current state

**Comment by julienlancelot** [comment]
@sns-seb Feedback apply ! (I've also re-introduced previously removal of toEntries())

**Comment by sns-seb** [comment]
you decide to take my last comment into account or not and then LGTM

**Pull request merged**